src/libYARP_init/src/CustomInit.cpp

```diff
@@ -48,6 +62,10 @@ yarp::os::Network::~Network() {
 void yarp::os::Network::init() {
     if (__custom_yarp_is_initialized==0) {
         initMinimum();
+        // If we init the clock inside the initMinum, it will loop into itself twice
+        // calling again the initMinimun, ending with __yarp_is_initialized counter 
+        // increased twice.
+        yarpClockInit();
```

Comment:
The problem with initializing the clock in the initMinimum is that the network clock requires to open a port, which requires yarp to be already initialized, as stated in the comment.

Placing the initialization inside the initMinimum generates the following flow:
`initMinimum` calls `initClock`, which requires to open a port, which in turns call again the `initMinimum` which calls again the `initClock`. For some reason, the second time the `initClock`  was called, it found yarp initialized and does its job, but I don't like that each yarp init calls `initMinimum` twice and more i don't like the `initMinimum` calling itself. It is not fully clear to me why it does not loop forever therefore I don't trust this mechanism so much.

I think that if someone calls InitMinimum and not the 'normal' init probably he/she knows what is doing and if the _network_ clock is needed he/she will initialize it.

In the final version the default clock will be the system clock therefore no explicit initialization will be required. ( I could do it also here... but this way of initializing stuff and having the clock=NULL as invalid helps finding applications in which there is something wrong with the usage of clock) 

If the clock initialization must be in the initMinimum I'll dig deeper in the problem of double initialization and find out a better way to initialize everything correctly.